### PR TITLE
If the container exit file is missing default exit code to -1

### PR DESCRIPTION
**Pull request by rhatdan**
If I create a sandbox pod and then restart the ocid service, the
pod ends up in a stopped state without an exit file.  Whether this is
a bug in ocid or not we should handle this case where a container exits
so that we can clean up the container.

This change just defaults to exit code to -1 if the container is not
running and does not have an exit file.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
I can cause this issue by doing the following

```
# systemctl start ocid
# POD_ID=$(sudo ocic pod ^Cn --config test/testdata/sandbox_config2.json)
# systemctl restart ocid
 ```
I end up with a journal entry like

```
Mar 14 16:27:04 dhcp-10-19-62-196.boston.devel.redhat.com ocid[18271]: time="2017-03-14 16:27:04.542057104-04:00" level=warning msg="could not restore sandbox ef0a120fd799392b7e01bdf63522226c3b39c0ef1183b85ff88b55d054bcfd0e container ef0a120fd799392b7e01bdf63522226c3b39c0ef1183b85ff88b55d054bcfd0e: error updating status for pod sandbox infra container ef0a120fd799392b7e01bdf63522226c3b39c0ef1183b85ff88b55d054bcfd0e: failed to find container exit file: stat /var/run/containers/overlay2-containers/ef0a120fd799392b7e01bdf63522226c3b39c0ef1183b85ff88b55d054bcfd0e/userdata/exit: no such file or directory"
```
Now I have no way to cleanup the container.  I have mounted overlay stuff all over. With this change I can see the pod and remove it using ocic.

File: oci/oci.go [hunk 1]
```diff
@@ -410,20 +410,21 @@ func (r *Runtime) UpdateStatus(c *Container) error {
...
-		c.state.Finished = time.Unix(st.Ctim.Sec, st.Ctim.Nsec)
-
-		statusCodeStr, err := ioutil.ReadFile(exitFilePath)
-		if err != nil {
-			return fmt.Errorf("failed to read exit file: %v", err)
-		}
-		statusCode, err := strconv.Atoi(string(statusCodeStr))
-		if err != nil {
-			return fmt.Errorf("status code conversion failed: %v", err)
+			c.state.ExitCode = -1
```
Comment (mrunalp) [inline oci/oci.go]: I think we should still log for this case.

Comment (rhatdan) [inline oci/oci.go]: Fixed

**Comment by runcom** [comment]
Travis timed out, not sure it's related, restarted

**Comment by mrunalp** [comment]
I think there is a real test issue here. Reproducible locally.

**Comment by runcom** [comment]
The first test failing is #6 which fails with https://travis-ci.org/kubernetes-incubator/cri-o/jobs/212232006#L2709

**Comment by rhatdan** [comment]
@runcom are you saying that https://github.com/kubernetes-incubator/cri-o/pull/6 is the problem?

**Comment by runcom** [comment]
No sorry, test number 6 not the PR number 6

**Comment by rhatdan** [comment]
Yes, not sure why this is breaking it though.

File: oci/oci.go [hunk 2]
```diff
@@ -410,20 +410,22 @@ func (r *Runtime) UpdateStatus(c *Container) error {
...
-		if err != nil {
-			return fmt.Errorf("status code conversion failed: %v", err)
+			logrus.Warnf("failed to find container exit file: %v", err)
+			c.state.ExitCode = -1
+		} else {
+			st := fi.Sys().(*syscall.Stat_t)
+			c.state.Finished = time.Unix(st.Ctim.Sec, st.Ctim.Nsec)
+
+			statusCodeStr, err := ioutil.ReadFile(exitFilePath)
+			if err == nil {
```
Comment (mrunalp) [inline oci/oci.go]: This is the issue. This should be `if err != nil {`

**Comment by rhatdan** [comment]
Can't see the forest from the trees.

Thanks.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**